pkg/cloud/gcp/actuators/machine/actuator.go

```diff
@@ -93,7 +128,12 @@ func (a *Actuator) Delete(ctx context.Context, cluster *clusterv1.Cluster, machi
 		machine:       machine,
 	})
 	if err != nil {
-		return fmt.Errorf(scopeFailFmt, machine.Name, err)
+		fmtErr := fmt.Errorf(scopeFailFmt, machine.Name, err)
+		return a.handleMachineError(machine, apierrors.InvalidMachineConfiguration("error decoding MachineProviderConfig: %v", fmtErr), deleteEventAction)
```

Comment:
`s/error decoding MachineProviderConfig/error creating machine scope` would be more appropriate

Reply:
InvalidMachineConfiguration does not seem appropriate here